thriftpool/list.go

```diff
@@ -0,0 +1,129 @@
+package thriftpool
+
+import (
+	"sync"
+)
+
+type listPool struct {
+	lock           sync.Mutex
+	pool           []Client
+	opener         ClientOpener
+	numActive      int32
+	minConnections int
+	maxConnections int
+}
+
+// Make sure listPool implements Pool interface.
+var _ Pool = (*listPool)(nil)
+
+// NewListPool creates a new thrift client pool implemented via list (slice).
+//
+// In most cases you should use the channel implementation instead.
```

Comment:
Is there a reason someone would use it?  It might make sense to only do the `channelPool` implementation if not, and if so a comment with some guidance about when you would use this might be nice.

Reply:
In the case that they might call `Release` after `Close`, maybe? But yea I don't feel strongly to keep this implementation.